u2-routing-controllers/features/productController.js

```diff
@@ -0,0 +1,20 @@
+
+const { findProducts } = require('../_services/fakedb')
+
+const getProduct = (req, res) => {
+  const productsName = req.query.q
+  const products = findProducts(productsName)
+  const productsPage = req.query.page || 1
+
+  if (products) {
```

Comment:
ในการใช้งานจริง เราจะทำสิ่งนี้ (ใช้พารามิเตอร์เส้นทางเพื่อค้นหาข้อมูลเพิ่มเติม). ใน U2 เป้าหมายของฉันคือเพื่อให้แน่ใจว่าคุณรู้วิธีดึงพารามิเตอร์เส้นทาง ฉันจะมีความสุขกับ `if (productsName)` รหัสของคุณล้ำหน้ากว่า :)

In real-world applications, we would do this (use the route parameters to lookup more information). In U2 my goal is just to make sure you know how to retrieve the route parameters. I would have been happy with `if (productsName)`. Your code is more advanced :)